sql/core/src/test/scala/org/apache/spark/sql/execution/BaseScriptTransformationSuite.scala

```diff
@@ -0,0 +1,390 @@
+            AttributeReference("value", StringType)()),
+          child = child,
+          ioschema = defaultIOSchema.copy(schemaLess = true)
+        ),
+        df.select(
+          'a.cast("string").as("key"),
+          concat_ws("\t",
+            'b.cast("string"),
+            'c.cast("string"),
+            decimalToString('d),
+            'e.cast("string"))).collect())
+    }
+  }
+
+  test("SPARK-30973: TRANSFORM should wait for the termination of the script (no serde)") {
+    assume(TestUtils.testCommandAvailable("/bin/bash"))
+
+    val rowsDf = Seq("a", "b", "c").map(Tuple1.apply).toDF("a")
+    val e = intercept[SparkException] {
+      val plan =
+        createScriptTransformationExec(
+          input = Seq(rowsDf.col("a").expr),
+          script = "some_non_existent_command",
+          output = Seq(AttributeReference("a", StringType)()),
+          child = rowsDf.queryExecution.sparkPlan,
+          ioschema = defaultIOSchema)
+      SparkPlanTest.executePlan(plan, spark.sqlContext)
+    }
+    assert(e.getMessage.contains("Subprocess exited with status"))
+    assert(uncaughtExceptionHandler.exception.isEmpty)
+  }
+
+  test("SPARK-32106: TRANSFORM should support more data types (no serde)") {
+    assume(TestUtils.testCommandAvailable("python"))
+    case class Struct(d: Int, str: String)
+    withTempView("v") {
+      val df = Seq(
+        (1, "1", 1.0, 11.toByte, BigDecimal(1.0), new Timestamp(1),
+          new Date(2020, 7, 1), new CalendarInterval(7, 1, 1000), Array(0, 1, 2),
+          Map("a" -> 1), new TestUDT.MyDenseVector(Array(1, 2, 3)), new SimpleTuple(1, 1L)),
+        (2, "2", 2.0, 22.toByte, BigDecimal(2.0), new Timestamp(2),
+          new Date(2020, 7, 2), new CalendarInterval(7, 2, 2000), Array(3, 4, 5),
+          Map("b" -> 2), new TestUDT.MyDenseVector(Array(1, 2, 3)), new SimpleTuple(1, 1L)),
+        (3, "3", 3.0, 33.toByte, BigDecimal(3.0), new Timestamp(3),
+          new Date(2020, 7, 3), new CalendarInterval(7, 3, 3000), Array(6, 7, 8),
+          Map("c" -> 3), new TestUDT.MyDenseVector(Array(1, 2, 3)), new SimpleTuple(1, 1L))
+      ).toDF("a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l")
+        .select('a, 'b, 'c, 'd, 'e, 'f, 'g, 'h, 'i, 'j, 'k, 'l,
+          struct('a, 'b).as("m"), unhex('a).as("n"), lit(true).as("o")
+        )
+      // Note column d's data type is Decimal(38, 18)
+      df.createTempView("v")
+
+      assert(spark.table("v").schema ==
+        StructType(Seq(
+          StructField("a", IntegerType, false),
+          StructField("b", StringType, true),
+          StructField("c", DoubleType, false),
+          StructField("d", ByteType, false),
+          StructField("e", DecimalType(38, 18), true),
+          StructField("f", TimestampType, true),
+          StructField("g", DateType, true),
+          StructField("h", CalendarIntervalType, true),
+          StructField("i", ArrayType(IntegerType, false), true),
```

Comment:
I get confused because you add tests for unsupported cases in `TRANSFORM should support more data types (no serde)`. https://github.com/apache/spark/pull/29085#discussion_r456992984

What's a purpose of this case in this test?

Reply:
Could you add separated tests for the unspported case?

Reply:
> I get confused because you add tests for unsupported cases in `TRANSFORM should support more data types (no serde)`. [#29085 (comment)](https://github.com/apache/spark/pull/29085#discussion_r456992984)
> 
> What's a purpose of this case in this test?

This test case can show  new added default spark row format serde can support all kind of data types and supported  output data types. (Complex type not support as output now)

Reply:
> Could you add separated tests for the unsupported case?

Doning this

Reply:
> This test case can show new added default spark row format serde can support all kind of data types and supported output data types. (Complex type not support as output now)

If so, could you make the test title more clearer?

Reply:
@maropu Seems some pr make hive serde support ArrayType/MapType/StructType.

Reply:
> If so, could you make the test title more clearer?

Updated